File: CCC-school-work/comporg/pnc2/ccattell/python/primeregbs.py
#!/usr/bin/python
import sys
import math
import time

start_time = time.time()

inputValue=int(sys.argv[1])
testNum = 4
primeCount = 2
divisor = 2
primeCheck = 0
##rem = 0
##root = 0

sys.stderr.write("2 3 ")

while primeCount < inputValue:
	root = int(math.sqrt(testNum))

    ##############################################
	while divisor <= root:
		rem = testNum % divisor
##		print(rem)
		if rem == 0:
			primeCheck = 1
			break
		else: 
			primeCheck = 0
			divisor = divisor + 1
    #############################################

##	primeCount = primeCount + 1

	if primeCheck == 0:
		sys.stderr.write("%i " % testNum)
		primeCount = primeCount + 1

	testNum = testNum + 1
	divisor = 2
	
##	if primeCount == inputValue:
##		sys.exit()		

sys.stderr.write("\n")
totalTime = time.time() - start_time
print "%.3f" % totalTime

File: CCC-school-work/comporg/pnc2/ccattell/perl/primeregbs.pl
#!/usr/bin/perl

use Time::HiRes;


my $start_run = Time::HiRes::gettimeofday();

$input = $ARGV[0];
$testNum = 4;
$divisor = 2;
$primeCount = 2;
$primeCheck = 0;
$root = 0;

printf STDERR '2 3 ';

while ($primeCount < $input){

	$root = sqrt($testNum);

	while ($divisor < $root + 1){
		if (($testNum % $divisor) == 0){
			$primeCheck = 1;
			break;
		}
		$divisor++;
	}

	if ($primeCheck == 0){
		printf STDERR '%d ', $testNum;
		$primeCount++;
	}
	$divisor = 2;
	$testNum++;
	$primeCheck = 0;
	
}

print "\n";

my $end_run = Time::HiRes::gettimeofday();
my $run_time = ($end_run - $start_run) / 1000;
printf("%.3f\n", $end_run - $start_run);


